internal/search/codeownership/ruleset.go

```diff
@@ -0,0 +1,80 @@
+package codeownership
+
+import (
+	"context"
+	"strings"
+
+	"github.com/hmarr/codeowners"
+
+	"github.com/sourcegraph/sourcegraph/internal/api"
+	"github.com/sourcegraph/sourcegraph/internal/authz"
+	"github.com/sourcegraph/sourcegraph/internal/gitserver"
+)
+
+type Ruleset struct {
+	codeownersRuleset codeowners.Ruleset
+}
+type Owners = []codeowners.Owner
+
+func (r *Ruleset) Match(path string) (Owners, error) {
+	if r.codeownersRuleset == nil {
+		return Owners{}, nil
+	}
+
+	rule, err := r.codeownersRuleset.Match(path)
+	if err != nil {
+		return Owners{}, err
+	}
+
+	if rule == nil {
+		return Owners{}, nil
+	}
+
+	return rule.Owners, nil
+}
+
+func NewRuleset(ctx context.Context, gitserver gitserver.Client, repoName api.RepoName, commitID api.CommitID) (Ruleset, error) {
+	var content []byte
+	var err error
+
+	ruleset := Ruleset{}
+
+	content, err = loadOwnershipFile(ctx, gitserver, repoName, commitID)
```

Comment:
nit: why declare these ahead of time?

```suggestion
	ruleset := Ruleset{}

	content, err := loadOwnershipFile(ctx, gitserver, repoName, commitID)
```